theano/tensor/nnet/corr_gemm.c

```diff
@@ -224,6 +217,8 @@ PyArrayObject* corrMM(PyArrayObject* bottom,
                     bottomWidth, kH, kW, padH, padW, dH, dW,
                     (%(float_type)s*)PyArray_DATA(col)+ tid * col_stride);
             // Second, gemm
+            // Always forcing gemm to one thread here for best and stable performance.
+            %(blas_flags)s;
```

Comment:
How to restore to the previous value after the call to gemm? This is to allow BLAS to be parallel for other layers like dense layer that do just 1 big gemm call.

Also, I would call that before the for loop. Idem for the 3 other places.